spectral_cube/tests/test_projection.py

```diff
@@ -141,6 +142,87 @@ def test_self_arith(LDO, data):
     assert p2.wcs == p.wcs
     assert np.all(p2.value==0)
 
+
+@pytest.mark.parametrize(('LDO', 'data'),
+                         zip(LDOs, data_twelve))
+def test_self_arith_with_beam(LDO, data):
+
+    exp_beam = Beam(1.0 * u.arcsec)
+
+    image = data
+    p = LDO(image, copy=False)
+    p = p.with_beam(exp_beam)
+
+    p2 = p + p
+
+    assert hasattr(p2, '_wcs')
+    assert p2.wcs == p.wcs
+    assert np.all(p2.value==2)
+    assert p2.beam == exp_beam
+
+    p2 = p - p
+
+    assert hasattr(p2, '_wcs')
+    assert p2.wcs == p.wcs
+    assert np.all(p2.value==0)
+    assert p2.beam == exp_beam
+
+
+@pytest.mark.xfail(raises=ValueError, strict=True)
+def test_VRODS_wrong_beams_shape():
+    '''
+    Check that passing Beams with a different shape than the data
+    is caught.
+    '''
+    exp_beams = Beams(np.arange(1, 4) * u.arcsec)
+
+    p = VaryingResolutionOneDSpectrum(twelve_qty_1d, copy=False,
+                                      beams=exp_beams)
+
+
+def test_VRODS_with_beams():
+
+    exp_beams = Beams(np.arange(1, twelve_qty_1d.size + 1) * u.arcsec)
+
+    p = VaryingResolutionOneDSpectrum(twelve_qty_1d, copy=False, beams=exp_beams)
+    assert (p.beams == exp_beams).all()
+
+    p = p.with_beams(exp_beams * 2)
```

Comment:
is it a mistake that we're allowing beams to be multiplied?  I don't think that operation makes sense, and if you try to multiply a single beam by a value, it fails:
```
In [11]: bm*5
Traceback (most recent call last):
  File "<ipython-input-11-9c711351f2a3>", line 1, in <module>
    bm*5
  File "/Users/adam/repos/radio_beam/build/lib.macosx-10.7-x86_64-3.6/radio_beam/beam.py", line 345, in __mul__
    return self.convolve(other)
  File "/Users/adam/repos/radio_beam/build/lib.macosx-10.7-x86_64-3.6/radio_beam/beam.py", line 338, in convolve
    new_major, new_minor, new_pa = convolve(self, other)
  File "/Users/adam/repos/radio_beam/build/lib.macosx-10.7-x86_64-3.6/radio_beam/utils.py", line 126, in convolve
    (other.major * np.cos(other.pa))**2 +
AttributeError: 'int' object has no attribute 'major'
```

Reply:
Agreed. I'll just generate a new set of beams here.